apps/api/src/app/widgets/usecases/get-subscriber-template-preference/get-subscriber-template-preference.usecase.ts

```diff
@@ -61,10 +64,9 @@ export class GetSubscriberTemplatePreference {
   }
 }
 
-function filterActiveChannels(channels: IPreferenceChannels, activeChannels: ChannelTypeEnum[]): IPreferenceChannels {
-  const filteredChannels = Object.assign({}, channels);
-
-  for (const key in channels) {
+function getDefaultFromTemplate(activeChannels: ChannelTypeEnum[], defaults: IPreferenceChannels) {
```

Comment:
Maybe worth returning here a particular type: `IPreferenceChannels` or something like this